ts/components/screens/BaseScreenComponent/index.tsx

```diff
@@ -0,0 +1,234 @@
+  headerBody?: React.ReactNode;
+  headerBackgroundColor?: ColorValue;
+  appLogo?: boolean;
+  searchType?: SearchType;
+  reportAttachmentTypes?: DefaultReportAttachmentTypeConfiguration;
+
+  // As of now, the following prop is propagated through 4 levels
+  // to finally display a checkbox in SendSupportRequestOptions
+  shouldAskForScreenshotWithInitialValue?: boolean;
+}
+
+export type Props = OwnProps &
+  ComponentProps<typeof BaseHeader> &
+  Pick<ComponentProps<typeof ContextualHelp>, "faqCategories">;
+
+const maybeDark = fromPredicate(
+  (isDark: boolean | undefined = undefined) => isDark === true
+);
+
+const ANDROID_OPEN_REPORT_DELAY = 50 as Millisecond;
+
+const contextualHelpModalAnimation = Platform.select<
+  ModalBaseProps["animationType"]
+>({
+  ios: "slide",
+  default: "none"
+});
+
+const BaseScreenComponentFC: React.FC<Props> = ({
+  accessibilityEvents,
+  accessibilityLabel,
+  appLogo,
+  children,
+  contextualHelp,
+  contextualHelpMarkdown,
+  customGoBack,
+  customRightIcon,
+  dark,
+  faqCategories,
+  goBack,
+  headerBackgroundColor,
+  headerBody,
+  headerTitle,
+  isSearchAvailable,
+  onAccessibilityNavigationHeaderFocus,
+  primary,
+  reportAttachmentTypes,
+  shouldAskForScreenshotWithInitialValue,
+  showInstabugChat,
+  titleColor
+}) => {
+  const currentScreenName = useSelector(
+    (store: GlobalState) => getCurrentRouteName(store.nav) || "not-found"
+  );
+
+  const [isHelpVisible, setIsHelpVisible] = useState(false);
+  // if the content is markdown we listen for load end event, otherwise the content is
+  // assumed always loaded
+  const [markdownContentLoaded, setMarkdownContentLoaded] = useState<boolean>(
+    !contextualHelpMarkdown
+  );
+
+  // used to trigger the side-effect base on timeout to take the screenshot
+  const [requestAssistanceData, setRequestAssistanceData] = useState<{
+    payload: RequestAssistancePayload;
+    attachmentConfig: DefaultReportAttachmentTypeConfiguration;
+  } | null>(null);
+
+  useEffect(() => {
+    if (requestAssistanceData) {
+      // since in Android we have no way to handle Modal onDismiss event https://reactnative.dev/docs/modal#ondismiss
+      // we force handling here. The timeout is due to wait until the modal is completely hidden
+      // otherwise in the Instabug screeshot we will see the contextual help content instead the screen below
+      // TODO: To complete the porting to 0.63.x, both iOS and Android will use the timeout. https://www.pivotaltracker.com/story/show/174195300
+      setTimeout(() => {
+        handleOnContextualHelpDismissed(
+          requestAssistanceData.payload,
+          requestAssistanceData.attachmentConfig
+        );
+        setRequestAssistanceData(null);
+      }, ANDROID_OPEN_REPORT_DELAY);
+    }
+  }, [requestAssistanceData]);
+
+  const showHelp = () => {
+    void mixpanelTrack("OPEN_CONTEXTUAL_HELP", {
+      SCREEN_NAME: noAnalyticsRoutes.has(currentScreenName)
+        ? "_"
```

Comment:
I have some doubts about tracking this "don't care"
Maybe it could reveal some implicit info since we have the navigation history and in addition could it be useful for our purposes?

Reply:
Assuming that we want to use this event to understand what are the most difficult screens for the users, an event without the screen doesn't help much.
Besides the privacy considerations, I think it make sense for us to skip the event altogether when the screen is blacklisted.

Reply:
I've applied the logic in b590954a6 to skip the event in case the route is in blacklist.